File: app/src/main/java/com/atom/vpn/demo/fragment/MainFragment.java
/*
 * Copyright (c) 2018 ATOM SDK Demo.
 * All rights reserved.
 */
package com.atom.vpn.demo.fragment;

import android.content.Intent;
import android.os.Bundle;
import android.os.Handler;
import android.text.TextUtils;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.EditText;
import android.widget.ImageView;
import android.widget.LinearLayout;
import android.widget.Switch;
import android.widget.TextView;
import android.widget.Toast;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.fragment.app.Fragment;

import com.atom.vpn.demo.AtomDemoApplicationController;
import com.atom.vpn.demo.R;
import com.atom.vpn.demo.activity.ConnectActivity;
import com.atom.vpn.demo.common.Constants;
import com.tooltip.Tooltip;

import static com.atom.vpn.demo.common.Utilities.setViewAndChildrenEnabled;

import de.blinkt.openvpn.BuildConfig;


public class MainFragment extends Fragment {

    private static final String TAG = "MainFragment";
    private LinearLayout usernameLay;
    private LinearLayout passwordLay;
    private EditText etUsername;
    private EditText etPassword;

    public MainFragment() {
        // Required empty public constructor
    }

    @Override
    public void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);

    }

    @Override
    public View onCreateView(@NonNull LayoutInflater inflater, ViewGroup container,
                             Bundle savedInstanceState) {
        // Inflate the layout for this fragment
        return inflater.inflate(R.layout.fragment_main, container, false);
    }

    @Override
    public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceState) {
        super.onViewCreated(view, savedInstanceState);
        LinearLayout secretKeyLay = view.findViewById(R.id.secretKeyLay);
        Handler mHandler = new Handler();
        mHandler.postDelayed(() -> secretKeyLay.setAlpha(0.5f), 500);


        TextView secretKey =  view.findViewById(R.id.secretKey);
        if (TextUtils.isEmpty(getString(R.string.atom_secret_key))) {
            secretKey.setText(getString(R.string.empty_secret_key));
        } else {
            secretKey.setText(getString(R.string.atom_secret_key));
        }

        ((TextView) view.findViewById(R.id.sdkVersion)).setText("Version: "+ BuildConfig.SDKVersion);

        usernameLay = view.findViewById(R.id.usernameLay);
        passwordLay =  view.findViewById(R.id.passwordLay);

        setViewAndChildrenEnabled(usernameLay, true);
        setViewAndChildrenEnabled(passwordLay, true);

        etUsername =  view.findViewById(R.id.etUsername);
        etPassword =  view.findViewById(R.id.etPassword);

        // Connect with Params
        view.findViewById(R.id.btnConnectParams).setOnClickListener(v -> {
            launchActivityForConnectionMethod(1);
        });


        // Connect with Dedicated IP
        view.findViewById(R.id.btnConnectDedicatedIp).setOnClickListener(v -> {
            launchActivityForConnectionMethod(2);
        });


        // Connect with Channel
        view.findViewById(R.id.btnConnectChannel).setOnClickListener(v -> {
            launchActivityForConnectionMethod(3);
        });
    }

    private void launchActivityForConnectionMethod(int connectionMethodType){
        if (TextUtils.isEmpty(etUsername.getText().toString().trim())) {
            etUsername.setError(Constants.UsernameRequired);
        } else if (TextUtils.isEmpty(etPassword.getText().toString().trim())) {
            etUsername.setError(null);
            etPassword.setError(Constants.PasswordRequired);
        } else {

            etUsername.setError(null);
            etPassword.setError(null);

            if (AtomDemoApplicationController.getInstance().getAtomManager() != null) {

                Intent intent = new Intent(getActivity(), ConnectActivity.class);
                intent.putExtra("connection_type", connectionMethodType);
                intent.putExtra("vpnUsername", etUsername.getText().toString());
                intent.putExtra("vpnPassword", etPassword.getText().toString());
                intent.addFlags(Intent.FLAG_ACTIVITY_SINGLE_TOP);
                startActivity(intent);
            } else {
                Toast.makeText(getActivity(), "Atom Manager is not initialized", Toast.LENGTH_LONG).show();
            }
        }
    }

}

File: app/src/main/java/com/atom/vpn/demo/fragment/MainFragment.kt
/*
 * Copyright (c) 2018 ATOM SDK Demo.
 * All rights reserved.
 */
package com.atom.vpn.demo.fragment

import android.content.Intent
import android.os.Bundle
import android.os.Handler
import android.os.Looper
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.EditText
import android.widget.LinearLayout
import android.widget.TextView
import android.widget.Toast
import androidx.fragment.app.Fragment
import com.atom.vpn.demo.R
import com.atom.vpn.demo.activity.ConnectActivity
import com.atom.vpn.demo.atom
import com.atom.vpn.demo.common.Constants
import com.atom.vpn.demo.common.Utilities.setViewAndChildrenEnabled
import de.blinkt.openvpn.BuildConfig

class MainFragment : Fragment() {

    private lateinit var etUsername: EditText
    private lateinit var etPassword: EditText

    override fun onCreateView(
        inflater: LayoutInflater,
        container: ViewGroup?,
        savedInstanceState: Bundle?
    ): View {
        // Inflate the layout for this fragment
        return inflater.inflate(R.layout.fragment_main, container, false)
    }

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)
        val secretKeyLay = view.findViewById<LinearLayout>(R.id.secretKeyLay)
        Handler(Looper.getMainLooper()).postDelayed({ secretKeyLay.alpha = 0.5f }, 500)

        val secretKey = view.findViewById<TextView>(R.id.secretKey)
        secretKey.text = getString(R.string.atom_secret_key).ifEmpty {
            getString(R.string.empty_secret_key)
        }

        view.findViewById<TextView>(R.id.sdkVersion).text = "Version: ${BuildConfig.SDKVersion}"

        val usernameLay = view.findViewById<LinearLayout>(R.id.usernameLay)
        val passwordLay = view.findViewById<LinearLayout>(R.id.passwordLay)

        setViewAndChildrenEnabled(usernameLay, true)
        setViewAndChildrenEnabled(passwordLay, true)

        etUsername = view.findViewById(R.id.etUsername)
        etPassword = view.findViewById(R.id.etPassword)

        // Connect with Params
        view.findViewById<View>(R.id.btnConnectParams).setOnClickListener {
            launchActivityForConnectionMethod(1)
        }

        // Connect with Dedicated IP
        view.findViewById<View>(R.id.btnConnectDedicatedIp).setOnClickListener {
            launchActivityForConnectionMethod(2)
        }

        // Connect with Channel
        view.findViewById<View>(R.id.btnConnectChannel).setOnClickListener {
            launchActivityForConnectionMethod(3)
        }
    }

    private fun launchActivityForConnectionMethod(connectionMethodType: Int) {
        if (etUsername.text.toString().trim().isEmpty()) {
            etUsername.error = Constants.UsernameRequired
        } else if (etPassword.text.toString().trim().isEmpty()) {
            etUsername.error = null
            etPassword.error = Constants.PasswordRequired
        } else {

            etUsername.error = null
            etPassword.error = null

            if (atom != null) {

                val intent = Intent(requireContext(), ConnectActivity::class.java)
                intent.putExtra("connection_type", connectionMethodType)
                intent.putExtra("vpnUsername", etUsername.text.toString())
                intent.putExtra("vpnPassword", etPassword.text.toString())
                intent.addFlags(Intent.FLAG_ACTIVITY_SINGLE_TOP)
                startActivity(intent)
            } else {
                Toast.makeText(
                    requireContext(),
                    "Atom Manager is not initialized",
                    Toast.LENGTH_LONG
                ).show()
            }
        }
    }
}
